go/vt/vtgate/vindexes/lookup_internal.go

```diff
@@ -158,15 +165,32 @@ func (lkp *lookupInternal) Create(vcursor VCursor, rowsColValues [][]sqltypes.Va
 }
 
 func (lkp *lookupInternal) createCustom(vcursor VCursor, rowsColValues [][]sqltypes.Value, toValues []sqltypes.Value, ignoreMode bool, co vtgatepb.CommitOrder) error {
-	if len(rowsColValues) == 0 {
-		// This code is unreachable. It's just a failsafe.
+	// Trim rows with null values
+	trimmedRowsCols := make([][]sqltypes.Value, 0, len(rowsColValues))
+	trimmedToValues := make([]sqltypes.Value, 0, len(toValues))
+nextRow:
+	for i, row := range rowsColValues {
+		for j, col := range row {
+			if col.IsNull() {
+				if !lkp.IgnoreNulls {
+					return fmt.Errorf("lookup.Create: input has null values: row: %d, col: %d", i, j)
+				}
+				continue nextRow
+			}
+		}
+		trimmedRowsCols = append(trimmedRowsCols, row)
+		trimmedToValues = append(trimmedToValues, toValues[i])
```

Comment:
On second thought, we do not need this loop. This can be checked while filling bindvars.

Reply:
For the first comment, they recommend pre-allocating if your estimates are fairly accurate, which is the case here.
For the second comment, we have to do the zero-length check after trimming, which isn't straightforward.
All this will make sense when I answer the copy question below.